modules/core/04-channel/keeper/events.go

```diff
@@ -292,3 +292,28 @@ func emitChannelUpgradeInitEvent(ctx sdk.Context, portID string, channelID strin
 		),
 	})
 }
+
+// emitChannelUpgradeOpenEvent emits a channel upgrade open event
+//
+//lint:ignore U1000 Ignore unused function temporarily for debugging
+func emitChannelUpgradeOpenEvent(ctx sdk.Context, portID string, channelID string, currentChannel types.Channel) {
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeChannelUpgradeOpen,
+			sdk.NewAttribute(types.AttributeKeyPortID, portID),
+			sdk.NewAttribute(types.AttributeKeyChannelID, channelID),
+			sdk.NewAttribute(types.AttributeKeyChannelState, currentChannel.State.String()),
+			// TODO: add flush status when enum/boolean is present
+			sdk.NewAttribute(types.AttributeCounterpartyPortID, currentChannel.Counterparty.PortId),
+			sdk.NewAttribute(types.AttributeCounterpartyChannelID, currentChannel.Counterparty.ChannelId),
+			sdk.NewAttribute(types.AttributeKeyUpgradeConnectionHops, currentChannel.ConnectionHops[0]),
+			sdk.NewAttribute(types.AttributeKeyUpgradeVersion, currentChannel.Version),
+			sdk.NewAttribute(types.AttributeKeyUpgradeOrdering, currentChannel.Ordering.String()),
```

Comment:
Happy to leave as is. I could also see it as a valid choice to now use the regular attribute keys such as `types.AttributeKeyVersion`

Reply:
Im pro reusing attributes since the type of message is given above

Reply:
i will open an issue to discuss this if we should refactor in the future!